src/octoclass.cpp

```diff
@@ -500,14 +511,25 @@ void OctoClass::FadeMemory(const double &rate) {  // rate at which this function
 
         // tree nodes that are unknown
         if (is_occ != tree_.isNodeOccupied(n)) {  // if it was occupied then disoccupied, delete node
-            tree_.deleteNode(key, it.getDepth());
+            tree_.deleteNode(key, depth);
         }
     }
 
     // Fade memory in inflated tree
     for (octomap::OcTree::leaf_iterator it = tree_inflated_.begin_leafs(),
                                        end = tree_inflated_.end_leafs();
                                        it != end; ++it) {
+        // This if statement is a workaround for a bug in octomap. For some reason
+        // after we delete all nodes from the octree we end up with the root node
+        // not being deleted. If we have only the root node available (which we can
+        // capure by checking the depth of the iterator), then we can safely clear
+        // the tree, as it is not a use case we need
+        const uint depth = it.getDepth();
```

Comment:
seems like this could be a function on its own, called every time we do something with the tree. It appears twice in the code.

Reply:
are you talking about getting the depth?
It appears twice because it is in two different `for` loops

Reply:
If you are talking about the whole chunk of code that is repeated, note that one of them clears `tree_`, and the other clears `tree_inflated_`.

Reply:
I think you could write a function `clearIfEmpty` or something that took a reference to a tree, checked its depth, and emptied it. It's a small optimization but it is more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)